docs/tutorials/vehicle_model_tutorial.md

```diff
@@ -0,0 +1,255 @@
+
+1. Copy the following contents to the `sdv_v1.rs`
+
+```rust
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+// Note: In the future this code should be generated from a DTDL spec.
+
+pub mod hmi {
+    pub mod show_notification {
+        pub const ID: &str = "dtmi:sdv:HMI:ShowNotification;1";
+        pub const NAME: &str = "Show Notification";
+        pub const DESCRIPTION: &str = "Show a notification on the HMI.";
+        pub mod request {
+            pub const ID: &str = "dtmi:sdv:HMI:ShowNotification::request;1";
+            pub const NAME: &str = "Notification";
+            pub const DESCRIPTION: &str = "The notification to show on the HMI.";
+            pub type TYPE = String;
+        }
+        pub mod response {
+            pub const ID: &str = "dtmi:sdv:HMI:ShowNotification::response;1";
+        }
+    }
+}
+
+1. Reference the `hvac.json` DTDL file that you have created in [1.5 DTDL properties](#15-dtdl-properties)
+
+1. Copy the following contents to the `sdv_v1.rs` Rust file
+
+```rust
+pub mod hvac {
+    pub mod ambient_air_temperature {
+        pub const ID: &str = "dtmi:sdv:HVAC:AmbientAirTemperature;1";
+        pub const NAME: &str = "AmbientAirTemperature";
+        pub const DESCRIPTION: &str = "The immediate surroundings air temperature (in Fahrenheit).";
+        pub type TYPE = i32;
+    }
+
+    pub mod is_air_conditioning_active {
+        pub const ID: &str = "dtmi:sdv:HVAC:IsAirConditioningActive;1";
+        pub const NAME: &str = "IsAirConditioningActive";
+        pub const DESCRIPTION: &str = "Is air conditioning active?";
+        pub type TYPE = bool;
+    }
+}
+```
+
+This `sdv_v1.rs` is a representation of the vehicle DTDL model with an HMI and an HVAC in code.
+
+In [1.2 Create HVAC and HMI interfaces](#12-create-hvac-and-hmi-interfaces), the `@id` field for the HMI and HVAC digital twin interfaces are `dtmi:sdv:HMI;1` and `dtmi:sdv:HVAC;1`, respectively. These DTMIs are constructed in the sdv_v1.rs file, which we created in step 2.
+
+The sdv_v1.rs file contains two main modules: `hmi` and `hvac`. The HMI module constructs the interface for `dtmi:sdv:HMI;1`, and the hvac module constructs the interface for `dtmi:sdv:HVAC;1`.
+
+In the HMI module, there is a `show_notification` submodule that represents the ShowNotification command in DTDL. It has an ID, NAME, and DESCRIPTION constant, which correspond to the `@id`, `name`, and `description` fields in the `hmi.json` DTDL file. The request and response submodules represent the request and response of the command.
+
+Similarly, in the HVAC module, there are two submodules: ambient_air_temperature and is_air_conditioning_active. These represent the `AmbientAirTemperature` and `IsAirConditioningActive` properties in the `hvac.json` DTDL. Each submodule has an ID, NAME, DESCRIPTION, and TYPE constant, which correspond to the `@id`, `name`, `description`, and `schema` fields in DTDL.

```

Comment:
These are not submodules.  A better term might be fragments or entities.

Reply:
I'm using the terms `module` and `submodule` to refer exclusively to the Rust code and not the DTDL itself. I think these terms provide more clarity on how the Rust code relates to the DTDL.